### PR TITLE
[MAINTENANCE] Rule-Based Profiler -- In ParameterBuilder insure that metrics are validated for conversion to numpy array (to avoid deprecation warnings)

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* We now check metrics outputs thoroughly for conditions that would preclude the output from being converted into `numpy.ndarray` in a way that does not result in deprecation warnings being printed.
* Fix to value set `ParameterBuilder` to receive sanitized array values.
* This was thoroughly tested in a Jupyter notebook.
* Existing tests pass.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-927
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/helpers/util.py [hunk 1]
```diff
@@ -488,7 +488,7 @@ def compute_quantiles(
     )
     return NumericRangeEstimationResult(
         estimation_histogram=np.histogram(a=metric_values, bins=NUM_HISTOGRAM_BINS)[0],
-        value_range=np.array([lower_quantile, upper_quantile]),
+        value_range=np.asarray([lower_quantile, upper_quantile]),
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/helpers/util.py:491]: Is this backwards compatible? We need to support 1.14.1 and above.

File: great_expectations/rule_based_profiler/types/attributed_resolved_metrics.py [hunk 2]
```diff
@@ -10,6 +10,50 @@
...
+
+        if isinstance(values, (list, tuple)):
+            value: MetricValue = values[0]
+            # Python "None" is illegal as candidate for conversion into "numpy.ndarray" type.
+            if value is None:
+                return True
+
+            # Pandas "DataFrame" and "Series" are illegal as candidates for conversion into "numpy.ndarray" type.
+            if isinstance(value, (pd.DataFrame, pd.Series, set)):
+                return True
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/attributed_resolved_metrics.py:31]: What's the reason behind checking `values[0]`? Aren't we performing this check by recursively calling the func on each element down in L46?

File: great_expectations/rule_based_profiler/types/attributed_resolved_metrics.py [hunk 3]
```diff
@@ -10,6 +10,50 @@
 from great_expectations.types.attributes import Attributes
 
 
+def _condition_metric_values(metric_values: MetricValues) -> MetricValues:
+    def _detect_illegal_array_type_or_shape(values: MetricValues) -> bool:
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/types/attributed_resolved_metrics.py:14]: ❤️

**Review by NathanFarmer — APPROVED** [review]
LGTM!

File: great_expectations/rule_based_profiler/helpers/util.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/helpers/util.py:491]: @cdkini Yes -- this method has been in `numpy` for years -- I do not anticipate any compatibility issues.  Thanks.

File: great_expectations/rule_based_profiler/types/attributed_resolved_metrics.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/types/attributed_resolved_metrics.py:31]: @cdkini This is just a way to check one of the elements to see what the type is.  If there is a more elegant way, I would be curious to know -- thanks!

**Pull request merged**